docker/Dockerfile

```diff
@@ -108,8 +119,17 @@ COPY synapse /synapse/synapse/
 # ... and what we need to `pip install`.
 COPY pyproject.toml README.rst /synapse/
 
+# Repeat of earlier build argument declaration, as this is a new build stage.
+ARG TEST_ONLY_IGNORE_POETRY_LOCKFILE
+
 # Install the synapse package itself.
-RUN pip install --prefix="/install" --no-deps --no-warn-script-location /synapse
+# Unless we haven't populated requirements.txt, we don't install any dependencies
+# as we should already have those from the previous `pip install` step.
```

Comment:
The double negative in the comments makes it tricky to understand. Can we flip the wording around?
```suggestion
# If we have populated requirements.txt, we don't install any dependencies
# as we should already have those from the previous `pip install` step.
```